sdk/include/opentelemetry/sdk/metrics/instrument.h

```diff
@@ -0,0 +1,230 @@
+     */
+    virtual void update(T value) override {
+        this->mu_.lock();
+        agg_->update(value);
+        this->mu_.unlock();
+    }
+    
+    /**
+     * Returns the aggregator responsible for meaningfully combining update values.
+     *
+     * @param none
+     * @return the aggregator assigned to this instrument
+     */
+    virtual nostd::shared_ptr<Aggregator<T>> GetAggregator() final { return agg_; }
+    
+private:
+    nostd::shared_ptr<Aggregator<T>> agg_;
+    int ref_ = 0;
+};
+
+template <class T>
+class SynchronousInstrument : public Instrument, virtual public metrics_api::SynchronousInstrument<T> {
+    
+public:
+    
+    SynchronousInstrument() = default;
+    
+    SynchronousInstrument(nostd::string_view name,
+                          nostd::string_view description,
+                          nostd::string_view unit,
+                          bool enabled,
+                          metrics_api::InstrumentKind kind):Instrument(name,description,unit,enabled,kind)
+    {}
+    
+    /**
+     * Returns a Bound Instrument associated with the specified labels.         * Multiples requests
+     * with the same set of labels may return the same Bound Instrument instance.
+     *
+     * It is recommended that callers keep a reference to the Bound Instrument
+     * instead of repeatedly calling this operation.
+     *
+     * @param labels the set of labels, as key-value pairs
+     * @return a Bound Instrument
+     */
+    virtual nostd::shared_ptr<metrics_api::BoundSynchronousInstrument<T>> bind(const trace::KeyValueIterable &labels) override {
+        return nostd::shared_ptr<BoundSynchronousInstrument<T>>();
+    }
+    
+    virtual void update(T value, const trace::KeyValueIterable &labels) override = 0;
+    
+    /**
+     * Checkpoints instruments and returns a set of records which are ready for processing.
+     * This method should only be called by the Meter Class as part of the export pipeline.
+     *
+     * @param none
+     * @return vector of Records which hold the data attached to this synchronous instrument
+     */
+    virtual std::vector<Record> GetRecords() = 0;
+    
+};
+
+// Helper functions for turning a trace::KeyValueIterable into a string
+inline void print_value(std::stringstream &ss,
+                        common::AttributeValue &value,
+                        bool jsonTypes = false)
+{
+    switch (value.index())
+    {
+        case common::AttributeType::TYPE_STRING:
+            if (jsonTypes)
+                ss << '"';
+            ss << nostd::get<nostd::string_view>(value);
+            if (jsonTypes)
+                ss << '"';
+            break;
+        default:
+            throw std::invalid_argument("Labels must be strings");
+            break;
+    }
+};
+
+// Utility function which converts maps to strings for better performance
+inline std::string mapToString(const std::map<std::string,std::string> & conv){
+    std::stringstream ss;
+    ss <<"{ ";
```

Comment:
There are several places in this PR where the code is not well-formatted (e.g. spacing, indentation, etc.)

Reply:
Sorry about that, for some reason these weren't an issue in my IDE.  I'll be sure to go over them carefully and take care of all the issues.

Reply:
went back over and tried to clean up most of these